newIDE/app/src/Utils/Language.spec.js

```diff
@@ -0,0 +1,91 @@
+// @flow
+
+jest.mock('../locales/LocalesMetadata', () => [
+  {
+    languageCode: 'pt_PT',
+    translationRatio: 0.46,
+  },
+  {
+    languageCode: 'pt_BR',
+    translationRatio: 0.95,
+  },
+  {
+    languageCode: 'en',
+    translationRatio: 0.04,
+  },
+  {
+    languageCode: 'es_ES',
+    translationRatio: 0.97,
+  },
+  {
+    languageCode: 'fr_FR',
+    translationRatio: 0.35,
+  },
+  {
+    languageCode: 'fr_CA',
+    translationRatio: 0.17,
+  },
+]);
+
+jest.mock('./Language', () => {
+  const originalModule = jest.requireActual('./Language');
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    getBrowserLanguageOrLocale: jest.fn(),
+  };
+});
+
+import { getBrowserLanguageOrLocale } from './Language';
+import { getInitialPreferences } from '../MainFrame/Preferences/PreferencesProvider';
+
+describe('PreferencesProvider', () => {
+  describe('getInitialPreferences', () => {
+    describe('Browser with language', () => {
+      test('return the best translated locale among the possible locales for this language, if good enough', () => {
+        // $FlowFixMe
+        getBrowserLanguageOrLocale.mockReturnValue('pt');
```

Comment:
OK, not hugely important, but if you want it to detect that it's a mock you can probably do at the beginning of your file:
```
const mockGetBrowserLanguageOrLocale = jest.fn();

jest.mock('./Language'...
...
   getBrowserLanguageOrLocale: mockGetBrowserLanguageOrLocale,
...
```
Then you can use the mockedFunction as a proper mock, instead of importing the real one (that is in fact a mock because you mocked the module)

Reply:
I had to use a slightly different version: `getBrowserLanguageOrLocale: () => mockGetBrowserLanguageOrLocale()`.

Here is the why: https://blog.bam.tech/developer-news/fix-jest-mock-cannot-access-before-initialization-error